hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -313,6 +327,11 @@ protected void doUnlock(long lockId) throws TException, InterruptedException {
   }
 
   static void validateTableIsIceberg(Table table, String fullName) {
+    if ("TRUE".equalsIgnoreCase(table.getParameters().get(TABLE_CREATION_FROM_HIVE))) {
+      // No check is needed. The table has been created from Hive (HiveIcebergMetaHook), and we are about to initialize
+      // the Iceberg metadata
+      return;
+    }
```

Comment:
Is this needed? `TABLE_TYPE_PROP` is set in the pre-commit hook. In fact, can we just use that instead of `TABLE_CREATION_FROM_HIVE`?

Reply:
`TABLE_TYPE_PROP` is set, but `METADATA_LOCATION_PROP` is not set, and it would throw the exception. We want to bail out of the checks only if this is indeed in the table creation phase. In other places I have tried to keep to the original behavior.

Reply:
Okay, I see. I think I would prefer to simplify this and make it so that the table is Iceberg if and only if the table property is there. If we don't have the metadata location, then we can set metadata to null and everything else should work like normal.

Reply:
Done. Had to change a single tests, but that's seems ok.